packages/router/src/grpc/view-protocol-server/ownned-position-ids.ts

```diff
@@ -0,0 +1,13 @@
+import type { Impl } from '.';
+import { servicesCtx } from '../../ctx';
+
+export const ownedPositionIds: Impl['ownedPositionIds'] = async function* (req, ctx) {
+  const services = ctx.values.get(servicesCtx);
+
+  const { indexedDb } = await services.getWalletServices();
+
+  const positionIds = indexedDb.getOwnedPositionIds(req.positionState, req.tradingPair);
```

Comment:
We should do parameter validation here:

```typescript
  if (!req.positionState) throw new Error('Missing position state in request');
  if (!req.tradingPair) throw new Error('Missing trading pair in request');

  const positionIds = indexedDb.getOwnedPositionIds(req.positionState, req.tradingPair);
  
```

Reply:
But `positionState` and `tradingPair` should not be required. 
If they are required, the client will have to brute force all combinations of trading pairs and state to get all positions

Reply:
https://github.com/penumbra-zone/penumbra/blob/a483d8e377ea6728f865d844f9adc4e7fe8f8827/crates/view/src/storage.rs#L1655 
In rust, they are also not required

Reply:
Ah, I misread the protos, thanks for correcting!